### PR TITLE
Feature/local dev

### PR DESCRIPTION
**Pull request by markcoatsworth**
# PR Type (Feature)

## Short Description
* Added a new system job runner, which invokes and manages the model_service as a system process instead of a Slurm job
* Reintroduced GPT2 as a simple testing model
* Removed all model_service config, which is now provided by the gateway as arguments
* Included new example .env for system workflows

### CODE REVIEW

File: web/Dockerfile [hunk 1]
```diff
@@ -5,7 +5,7 @@ FROM --platform=linux/amd64 python:3.8
 WORKDIR /app
 
 # Install apt packages
-RUN apt update && apt install -y ldap-utils
+RUN apt update && apt install -y ldap-utils vim telnet iputils-ping less iproute2
```
Comment (jwilles) [inline web/Dockerfile:8]: Can we avoid the dependency bloat in the container?

File: web/model_instances/routes.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import requests
 from flask import Blueprint, request, current_app, jsonify
-from flask_jwt_extended import jwt_required, get_jwt_identity
+from flask_jwt_extended import jwt_required, get_jwt, get_jwt_identity, verify_jwt_in_request
+from functools import wraps
```
Comment (jwilles) [inline web/model_instances/routes.py:4]: I don't think we use this after importing it.

File: web/models.py [hunk 3]
```diff
@@ -16,11 +17,20 @@
         "name": "OPT-175B",
         "description": "175B parameter version of the Open Pre-trained Transformer (OPT) model trained by Meta",
         "url": "https://huggingface.co/meta/opt-175B",
+        "path": "/ssd005/projects/llm/OPT-175B-mp32"
```
Comment (jwilles) [inline web/models.py:20]: The model path should be configured on the model service side. Including it here couples the gateway service intimately with the cluster and the model service. It bakes in knowledge about the cluster job launching and file system.

File: web/models.py [hunk 4]
```diff
@@ -64,16 +74,26 @@ def launch(self):
...
             self._model_instance.transition_to_state(ModelInstanceStates.LAUNCHING)
         except Exception as err:
-            current_app.logger.error(f"Job launch failed: {err}")
+            current_app.logger.error(f"Job launch for {self._model_instance.name} failed: {err}")
             self._model_instance.transition_to_state(ModelInstanceStates.FAILED)
 
+    def register(self, host: str):
+        # Only the system job scheduler can register models in pending state
+        # since launching happens immediately
+        if Config.JOB_SCHEDULER == "system":
```
Comment (jwilles) [inline web/models.py:87]: This check couples the model instance to the job scheduler. We have abstracted this via the model_service_client and should that logic in there.

File: web/Dockerfile (hunk 1 above)
Comment (markcoatsworth) [reply web/Dockerfile]: Done

File: web/model_instances/routes.py (hunk 2 above)
Comment (markcoatsworth) [reply web/model_instances/routes.py]: Removed

File: web/models.py (hunk 4 above)
Comment (markcoatsworth) [reply web/models.py]: Something weird was up here. My model kept dropping into FailedState and this change seemed to fix that. But I tried removing this code entirely and the model still comes up properly. So it must have been a different problem entirely. This code is gone now.

File: web/models.py (hunk 3 above)
Comment (markcoatsworth) [reply web/models.py:20]: As we discussed in person, this is a much bigger change that warrants its own ticket. Can we just leave this in here for now and then do the refactor later?

**Comment by jwilles** [comment]
LGTM

**Pull request merged**